docs/Tutorial_DiscordMERLIN.ipynb

```diff
@@ -0,0 +1,1911 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #51.        T_subseq_isfinite = core.rolling_isfinite(T, m)
</pre><p>I'm realizing that if this line is inside of the function then it implies that we must not do <code>core.preprocess(T, m)</code> before <code>_select_candidates</code> is called! This is not obvious and there is no way to enforce this so we should actually make <code>T_subseq_isfinite</code> an input parameter to <code>_select_candidates</code> . So, whatever outermost wrapper must NOT call <code>core.preprocess(T, m)</code> before calling <code>T_subseq_isfinite</code> since <code>core.preprocess</code> will convert all non-finite values to zero! So, technically, in your current version <code>T_subseq_isfinite</code> will never have <code>np.inf</code> since <code>T</code> has already been transformed!</p><p> </p><p>However, there's a simple trick. Whether or not something <code>isfinite</code> is technically captured in <code>M_T</code> !! Here is simple example:</p><p> </p><pre class="ql-syntax" spellcheck="false">from stumpy impore core
import numpy as np

T = np.random.rand(10)
T[8] = np.inf
T[1] = np.nan
core.preprocess(T, m=5)

(array([0.72331461, 0.&nbsp; &nbsp; &nbsp; &nbsp; , 0.8349656 , 0.17393654, 0.76926601,
&nbsp; &nbsp; &nbsp; &nbsp; 0.23217637, 0.27976504, 0.86053412, 0.&nbsp; &nbsp; &nbsp; &nbsp; , 0.10005755]),

&nbsp;array([&nbsp; &nbsp; &nbsp; &nbsp;inf,&nbsp; &nbsp; &nbsp; &nbsp; inf, 0.45802191, 0.46313562,&nbsp; &nbsp; &nbsp; &nbsp; inf,
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;inf]),

&nbsp;array([0.26363881, 0.3011338 , 0.28370606, 0.29060091, 0.2818241 ,
&nbsp; &nbsp; &nbsp; &nbsp; 0.29181176]))
</pre><p> </p><p>Notice that the middle array, <code>M_T</code> , accurately captures which rolling windows <code>isfinite</code> so you can simply do:</p><pre class="ql-syntax" spellcheck="false">T_subseqs_isfinite = np.isfinit(M_T)
</pre><p>Apologies for not realizing this earlier! You should do this in all of the functions that <code>T_subeqs_isfinite</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
> I'm realizing that if this line is inside of the function then it implies that we must not do `core.preprocess(T, m)` before `_select_candidates` is called!

Yes. I did not use  `core.preprocess(T, m)` though. In current version, I did not clean `T`, and I am just passing the `(raw) T` to the functions).

Did you mean we may make a mistake later by using `core.preprocess(T,m)` before calling `T_subseq_isfinite`? (so we should better take care of it now?)

> Notice that the middle array, `M_T` , accurately captures which rolling windows `isfinite` so you can simply do:

That is nice. If we decide to use  `core.preprocess(T, m)`  to avoid potential future mistakes, I can use this trick.

Reply:
Yes, we should just call `core.preprocess` as this is the safest thing to do